apps/server/src/modules/task/uc/task.authorization.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { CourseRepo } from '@src/repositories';
+
+export enum TaskParentPermission {
+	read,
+	write,
+}
+
+@Injectable()
+export class TaskAuthorizationService {
+	constructor(private readonly courseRepo: CourseRepo) {}
+
+	/**
+	 * Important user group operations are only a temporary solution until we have established groups
+	 */
+	async getPermittedCourses(userId: EntityId, permission: TaskParentPermission): Promise<EntityId[]> {
+		// coursegroups are missing
+		// lessons are missing -> only search for hidden: false,
+		const [permittedCourses] =
+			permission === TaskParentPermission.write
```

Comment:
for such important decisions i would not like the ?: shorthand as its a switch/case situation, why selecting teacher/student by write permission? the logic is not clear to me. we get courses by a task-permission while courses and tasks not know each other?!?!?

Reply:
i think we should actually decide by the user's role. in the future we could even add a parameter with the intended role to the use case. that would be if the user had more than one mathing role.
because we currently have no easy way of finding out the user's roles the current code (for the moment) sticks to the permissions to mimic the roles somewhat. finding the roles of the user has to be implemented asap.